### ISSUE TITLE
Enable Unix domain sockets to be used via System.Net.Sockets

### ISSUE THREAD

**Issue by stephentoub** [dotnet/corefx#4588]

**Comment by stephentoub** [comment]
Blocked by #4777

**Comment by stephentoub** [comment]
@IlyaBiryukov, I saw you made some progress on this recently (thanks!).  Has this been fully implemented, or is there more to be done?

**Comment by IlyaBiryukov** [comment]
The #4777 is fixed. You can use Unix domain sockets with custom descendant of EndPoint class.
See for example in https://github.com/dotnet/corefx/blob/master/src/System.Net.Sockets/tests/FunctionalTests/UnixDomainSocketTest.cs

Another question is if we want to have this class in CoreFX, like IPEndPoint and DnsEndPoint. I'd say it would be nice to have it, but that is not blocking and can wait for the next release.

**Comment by stephentoub** [comment]
Thanks, @IlyaBiryukov!  For now, making it easy to use as sample code someone can pull into their project is good enough.  We can consider introducing additional surface area in the future.
